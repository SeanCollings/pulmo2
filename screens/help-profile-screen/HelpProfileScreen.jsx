import React from 'react';
import { StyleSheet, Text, ScrollView } from 'react-native';
import MaterialCommunityIcons from 'react-native-vector-icons/MaterialCommunityIcons';

import { useTheme } from '../../hooks/useTheme';
import options from './options';

export const helpProfileScreenOptions = options;

const HelpProfileScreen = (props) => {
  const theme = useTheme();
  const styleTextContent = {
    ...styles.textContent,
    color: theme.TEXT,
    opacity: theme.DARK ? 0.87 : 1,
  };

  return (
    <ScrollView
      style={{ ...styles.container, backgroundColor: theme.BACKGROUND }}
      contentContainerStyle={{ alignItems: 'center' }}
    >
      <Text style={{ ...styles.textHeading, color: theme.SECONDARY }}>
        Profile Page
      </Text>
      <Text style={styleTextContent}>
        Use{' '}
        <MaterialCommunityIcons
          name="chevron-left-circle"
          color={theme.BORDER}
          size={20}
        />{' '}
        and{' '}
        <MaterialCommunityIcons
          name="chevron-right-circle"
          color={theme.BORDER}
          size={20}
        />{' '}
        to select the difficulty level that you will use on the Luft&trade;.
      </Text>
      <Text
        style={{
          ...styleTextContent,
          ...styles.textBold,
          ...(theme.DARK
            ? { color: theme.TERTIARY, fontFamily: 'tit-light' }
            : {}),
        }}
      >
        At the start of each excercise you will confirm the difficulty level.
      </Text>
      <Text style={styleTextContent}>
        Each excercise will be saved to your Profile Page and can be viewed in
        detail by selecting it from 'All activities'.
      </Text>
      <Text style={styleTextContent}>
        If you would like to delete an excercise, click through to an activity's
        details and select delete.
      </Text>
      <Text style={styleTextContent}></Text>
    </ScrollView>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    paddingTop: 30,
  },
  textHeading: {
    paddingBottom: 20,
    fontSize: 20,
    fontFamily: 'tit-regular',
  },
  textContent: {
    fontSize: 16,
    width: '80%',
    textAlign: 'center',
    paddingBottom: 20,
    fontFamily: 'tit-light',
  },
  textBold: {
    fontFamily: 'tit-bold',
  },
});

export default HelpProfileScreen;
