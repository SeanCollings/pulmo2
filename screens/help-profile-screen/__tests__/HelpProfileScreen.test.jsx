import React from 'react';
import { act, create } from 'react-test-renderer';
import HelpProfileScreen, {
  helpProfileScreenOptions,
} from '../HelpProfileScreen';

jest.useFakeTimers();
jest.mock('react-native/Libraries/Animated/src/NativeAnimatedHelper');

describe('HelpProfileScreen - unit test', () => {
  test('should render', async () => {
    await act(async () => {
      tree = create(<HelpProfileScreen />);
    });

    expect(tree).toMatchSnapshot();
  });

  describe('helpProfileScreenOptions', () => {
    test('options', () => {
      const options = helpProfileScreenOptions();
      expect(options).toMatchSnapshot();
    });

    test('headerTitle', () => {
      const headerTitle = helpProfileScreenOptions().headerTitle({
        tintColor: 'white',
      });
      expect(headerTitle).toMatchSnapshot();
    });
  });
});
