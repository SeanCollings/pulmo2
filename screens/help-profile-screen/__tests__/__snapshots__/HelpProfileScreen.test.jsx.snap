// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelpProfileScreen - unit test helpProfileScreenOptions headerTitle 1`] = `
<Text
  style={
    Object {
      "color": "white",
      "fontSize": 20,
    }
  }
>
  Help
</Text>
`;

exports[`HelpProfileScreen - unit test helpProfileScreenOptions options 1`] = `
Object {
  "headerTitle": [Function],
}
`;

exports[`HelpProfileScreen - unit test should render 1`] = `
<RCTScrollView
  contentContainerStyle={
    Object {
      "alignItems": "center",
    }
  }
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
      "paddingTop": 30,
    }
  }
>
  <View>
    <Text
      style={
        Object {
          "color": "#002f56",
          "fontFamily": "tit-regular",
          "fontSize": 20,
          "paddingBottom": 20,
        }
      }
    >
      Profile Page
    </Text>
    <Text
      style={
        Object {
          "color": "#3e4b57",
          "fontFamily": "tit-light",
          "fontSize": 17,
          "opacity": 1,
          "paddingBottom": 20,
          "textAlign": "center",
          "width": "80%",
        }
      }
    >
      Use
       
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "#CCC",
              "fontSize": 20,
            },
            undefined,
            Object {
              "fontFamily": "material-community",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        ﬍
      </Text>
       
      and
       
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "#CCC",
              "fontSize": 20,
            },
            undefined,
            Object {
              "fontFamily": "material-community",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        ﬏
      </Text>
       
      to select the difficulty level that you will be using on the Luft™.
    </Text>
    <Text
      style={
        Object {
          "color": "#3e4b57",
          "fontFamily": "tit-bold",
          "fontSize": 17,
          "opacity": 1,
          "paddingBottom": 20,
          "textAlign": "center",
          "width": "80%",
        }
      }
    >
      At the start of each excercise you will confirm the difficulty level.
    </Text>
    <Text
      style={
        Object {
          "color": "#3e4b57",
          "fontFamily": "tit-light",
          "fontSize": 17,
          "opacity": 1,
          "paddingBottom": 20,
          "textAlign": "center",
          "width": "80%",
        }
      }
    >
      Each excercise will be saved to your Profile and can be viewed in detail by selecting it from 'All activities'.
    </Text>
    <Text
      style={
        Object {
          "color": "#3e4b57",
          "fontFamily": "tit-light",
          "fontSize": 17,
          "opacity": 1,
          "paddingBottom": 20,
          "textAlign": "center",
          "width": "80%",
        }
      }
    >
      If you would like to delete or edit an excercise, click through to an activity's details and select your desired option.
    </Text>
    <Text
      style={
        Object {
          "color": "#3e4b57",
          "fontFamily": "tit-light",
          "fontSize": 17,
          "opacity": 1,
          "paddingBottom": 20,
          "textAlign": "center",
          "width": "80%",
        }
      }
    />
  </View>
</RCTScrollView>
`;
